astropy/table/table.py

```diff
@@ -3683,15 +3684,26 @@ def __eq__(self, other):
         return self._rows_equal(other)
 
     def __ne__(self, other):
-        return ~self.__eq__(other)
+        eq = self.__eq__(other)
+        if isinstance(eq, bool):
```

Comment:
Could do `return self.__eq__(other) == False`, but arguably trying to be too clever...

Reply:
Actually this would probably be more performant (`isinstance` is infamously slow). If this function is considered performance-critical in any way I think it's worth considering, otherwise I think it just hurts readability (I know it's hard to resist the call of golfing sometimes !)

Reply:
Indeed, readability is why I felt I was trying to be too clever. Let's stick with what you have!